src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -288,17 +289,19 @@ private HttpMessage CreateRequest(string uri)
             return message;
         }
 
-        private static bool IsFinalState(Response response, HeaderSource headerSource, out OperationState? failureState)
+        private static bool IsFinalState(Response response, HeaderSource headerSource, out OperationState? failureState, out string? resourceLocation)
         {
             failureState = null;
+            resourceLocation = null;
+
             if (headerSource == HeaderSource.Location)
             {
                 return response.Status != 202;
             }
 
             if (response.Status is >= 200 and <= 204)
             {
-                if (response.ContentStream?.Length > 0)
+                if (response.ContentStream is {CanSeek: true, Length: > 0})
```

Comment:
what if `CanSeek: false`? If `204` is returned, will the `ContentStrea.CanSeek == false`?

Reply:
If `CanSeek` is false, we won't be able to call `response.ContentStream.Position = 0;` 
It's a good question if we should throw here if `CanSeek` is false or do something else.

Reply:
Is there ever a scenario where we don't own constructing this?  It seems we should never hit this case and maybe adding the check only adds confusion?

In the past the seek would have thrown an exception it might be safest to simply let that continue to happen instead of changing the behavior?

Reply:
Resolved